News

Felotus opened the pull request:
переработал классы

Felotus committed:
new

Felotus committed:
Create forjs

Felotus committed:
new

Felotus committed:
Delete forjs

Felotus committed:
Create forclass

Felotus committed:
new

Felotus committed:
Delete forclass

subogdanov commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,308 @@
+
+	public function changeFieldVal($elem, $val, Field $field){
+		if ($field->get_type() !== $this->_field_types['multiselect']) {
+			$val = array('value' => $val);
```
Code Style. Посмотри еще раз как должны объявляться массивы

subogdanov commented on classes/Field.php:
```diff
@@ -0,0 +1,42 @@
+		return $this->_type;
+	}
+
+	public function get__elem_type(){
```
CS

subogdanov commented on classes/Acc.php:
```diff
@@ -0,0 +1,6 @@
+<?php
+	class Acc{
+		const ELEM_TYPE = 20;
```
Откуда взялся этот тип?

subogdanov commented on classes/Field.php:
```diff
@@ -0,0 +1,42 @@
+	private $_enums;
+	private $_origin;
+	private $_type;
+	const ELEM_TYPE = 30;
```
Тоже придумал?

Felotus committed:
updated

исправил массивы

subogdanov commented on classes/Company.php:
```diff
@@ -0,0 +1,27 @@
+	private $_contacts;
+	const ELEM_TYPE = 3;
+
+	public function __construct($name = NULL, $id = NULL, array $contacts = NULL){
```
Вынеси в отдельные сетеры. У тебя свойств может быть сколько угодно, и многие из них опциональны. Если тебе нужно будет только 10ое свойство обновить тут тебе все 9 придется перечислить.

Можно целые цепочки делать из сетеров, например:

```
<?php 
public function set_name($name) {
	$this->name = $name;
	return $this;
}

public function set_id($id) {
	$this->id = $id;
	return $this;
}
```

```
<?php 
$contact = new Contact();

$contact
	->set_id(1)
	->set_name('John doe');
```

Felotus committed:
upload __

исправил __

subogdanov commented on AmoConstruct.php:
```diff
@@ -0,0 +1,309 @@
+	private $_link;
+	private $_hash;
+	private $_max_row;
+	private $_elem_links = [1 => 'contacts',
```
здесь тоже поля летают

Felotus committed:
14.01

Заменил конструктор на сеттеры

Felotus committed:
Delete Acc.php

Felotus committed:
14.01

заменил конструктор на сеттеры

Felotus committed:
14.01

Поправил массивы, доп метод createElems, это просто тест, хочу как то 3 метода создания в 1 свернуть покрасивее

Felotus committed:
doctype

добавил doctype

Felotus committed:
phpdoc

Пробелы

Felotus committed:
Add files via upload

Обернул повторяющийся код при создании контактов, компаний, сделок и покупателей в отдельную функцию

Felotus committed:
Delete AmoConstruct.php

Felotus committed:
15.01

новая версия на ООП, вынес логику из AmoConstr

Felotus committed:
15.01

Вынес логику из AmoConstr

Felotus committed:
new

последние изменения на сегодня

Felotus committed:
new

последнее на сегодня

Felotus committed:
16.01 first

подправил код

Felotus committed:
Delete amo_mass_create.php

Felotus committed:
Delete event_add.php

Felotus committed:
Delete index.html

Felotus committed:
Delete task_add.php

Felotus committed:
Delete task_types_get.php

Felotus committed:
Delete task_close.php

Felotus committed:
Delete textfield_create.php

Felotus committed:
16.01

утрене изменение, дублирование кода или switch после for?

Felotus committed:
16.01

Утрене изменение, switch+for или дублирование кода?

Felotus committed:
16.01 last

Попробовал сделать одну точку входа, пока слабо понимаю как должно быть правильно в моем случае, но в одну точку свел

Felotus committed:
поправочка 

Попробовал сделать одну точку входа, пока слабо понимаю как должно быть правильно в моем случае, но в одну точку свел.
*поправочка

Felotus committed:
16.01 last

Попробовал сделать одну точку входа, пока слабо понимаю как должно быть правильно в моем случае, но в одну точку свел

Felotus committed:
16.01 last

Попробовал сделать одну точку входа, пока слабо понимаю как должно быть правильно в моем случае, но в одну точку свел

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+     * @param sting $hash
+     * @throws Exception
+     */
+    public function auth($akk, $mail, $hash){
```
acc тогда уж

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+    public function auth($akk, $mail, $hash){
+		$this->_mail = $mail;
+		$this->_hash = $hash;
+		$this->_link = "https://".$akk.".amocrm.ru/";
```
кодстайл

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+	public function createField($elem_type, Field $field){
+		$field_id = NULL;
+		$enums = NULL;
+		$data['add'][] = [
```
$data не инициализирована

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+				$elems[] = $elem;
+			}
+			return $elems;
+		} else {
```
Один ретерн и PHP док поправить, не мешай array и bool, сделай array | NULL

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+				$elems[] = $elem;
+			}
+			return $elems;
+		} else {
```
Тоже самое

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+					'values' => $values
+				];
+			}
+			$data['update'][] = [
```
$data не инициализирована

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+	 * @throws Exception
+	 */
+	public function createNote(AmoElem $elem, Note $note){
+		$data['add'][] = [
```
таже беда с $data

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+		];
+		$result = $this->post($this->_link.'api/v2/tasks', $data);
+		if (isset($result['_embedded']['items'][0]['id']) && $result['_embedded']['items'][0]['id'] === 0) {        
+			throw new Exception('не удалось найти элемент', 6);
```
поменять местами

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+		foreach ($elems as $k => $v) {
+			$data['add'][$k]['name'] = $elems[$k]->get_name();
+			switch (TRUE) {
+				case ($elem_type === Lead::ELEM_TYPE || $elem_type === Customer::ELEM_TYPE):
```
так задумано что нету break?

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+					break;
+			}
+		}
+		var_dump($data);
```
убрать

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+
+    /**
+     * @param sting $akk
+     * @param sting $mail
```
string

rmiaev commented on classes/AmoConstruct.php:
```diff
@@ -0,0 +1,457 @@
+     */
+    public function auth($akk, $mail, $hash){
+		$this->_mail = $mail;
+		$this->_hash = $hash;
```
зачем $this->_mail и $this->_hash если используешь в одном месте? в этом методе

rmiaev replied:
зачем $this->_mail и $this->_hash если используешь в одном месте? в этом методе

rmiaev commented on classes/CurlReq.php:
```diff
@@ -0,0 +1,62 @@
+<?php
+class CurlReq {
```
С кодстайлом беда во всем класс, пробелы забываешь

Felotus replied:
Это точно, я кодстайл читал, вроде стараюсь делать по нему, но не совсем понимаю, как правильно.

rmiaev commented on classes/Field.php:
```diff
@@ -0,0 +1,67 @@
+<?php
+class Field{
```
phpdoc

Felotus replied:
Я с ними не работал, постараюсь добавлять по мере возможностей, часто забываю, еще нужно почитать как правильно оформлять их, пока не знаю как точно подписывать коллекции объектов например

rmiaev commented on classes/Note.php:
```diff
@@ -0,0 +1,57 @@
+<?php
+class Note{
```
phpdoc

rmiaev commented on classes/Router.php:
```diff
@@ -0,0 +1,36 @@
+
+	//ниже тестовые методы
+	public function __construct(){
+		$routPath = $_SERVER['DOCUMENT_ROOT'].'/rout_config.php';
```
в трайкеч или include_once

rmiaev commented on classes/Router.php:
```diff
@@ -0,0 +1,36 @@
+		$this->_routes = include($routPath);
+	}
+
+	private function getURI(){
```
php doc, что вернет если !empty($_SERVER['REQUEST_URI']) будет false? надо как то обозначить

Felotus replied:
Уже изменил, я копипастил данный метод, сейчас посмотрел, он в принципе не мог вернуть пустое значение, как минимум возвращал просто /, исправил и false добавил уже

rmiaev commented on textfield_create.php:
```diff
@@ -0,0 +1,39 @@
+	$field_name = 'новое поле';
+	$elem_id = DataFilter::clear($_POST['id']);
+	$amo_us = new AmoConstruct($config['api']);
+	$amo_us->auth($config['akk'], $config['mail'], $config['hash'], $config['max_row']);
```
auth принимает 3 параметра, ты передаешь 4

rmiaev commented on textfield_create.php:
```diff
@@ -0,0 +1,39 @@
+	};
+	$elem->set_id($elem_id);
+	$field = new Field();
+	$field->set_type(Fied::TEXT);
```
нет такого класса Fied

Felotus replied:
Да увидел при проверке, исправил.

rmiaev commented on task_types_get.php:
```diff
@@ -0,0 +1,10 @@
+try {
+	$config = include('../config.php');
+	$amo_us = new AmoConstruct($config['api']);
+	$amo_us->auth($config['akk'], $config['mail'], $config['hash'], $config['max_row']);
```
Таже беда с параметрами

Felotus replied:
Исправил

rmiaev commented on task_types_get.php:
```diff
@@ -0,0 +1,10 @@
+<?php
+try {
+	$config = include('../config.php');
```
сделай config.php.example хоть глянуть что там

Felotus replied:
вынес

rmiaev commented on control.php:
```diff
@@ -0,0 +1,18 @@
+<?php
+spl_autoload_register(function ($className) {
+	$className = str_replace( "..", "", $className);
```
автолоадер в отдельный файл

Felotus replied:
Убрал, подумаю еще, может буду файлы в разные папки распихивать и из автолоадера сделаю класс какой

rmiaev commented on js/test.js:
```diff
@@ -0,0 +1,210 @@
+$(document).ready( function () {
```
ESlint поставь

Felotus replied:
Еще не поставил, почитаю изучу, куда его ставить нужно и как работать с ним

rmiaev commented on event_add.php:
```diff
@@ -0,0 +1,39 @@
+			break;
+	};
+	$elem->set_id($elem_id);
+	$note = new Note();
```
Если во всех сетах сделаешь return $this
то вызов будет вот такой 
$note->set_type(DataFilter::clear($_POST['note_type']))
	  ->set_val(DataFilter::clear($_POST['text']))
	  ->set_call_link('http://example.com/calls/1.mp3')
	  ->set_call_duration(30)
	  ->set_origin($config['origin']);

Felotus replied:
Блин забыл изменить в последней версии, к следующей добавлю

rmiaev commented on amo_mass_create.php:
```diff
@@ -0,0 +1,102 @@
+	$multi->set_enums($enums_val);
+	$multi->set_origin($config['origin']);
+	$multi = $amo_us->createField(Contact::ELEM_TYPE, $multi);
+	$multi = $amo_us->getFieldEnums(Contact::ELEM_TYPE, $multi);
```
судя по названию должно возвращать енамы, а возвращает инстанс Field, может лучше реализовать этот метод в классе Field, как и createField в принципе

Felotus replied:
Я просто хотел сделать максимально отстраненные классы, работой с АМО занимается этот класс, не хотел в носить в классы логику API, в основном классе на данный момент есть метод который добавляет пачками, ну и возвращает пачки объектов, саму логику работы API я бы все таки оставил в AmoConstruct, в методы общностей я думаю можно добавить методы которые получают объект конструктора и через него запускают его соответствующий метод, например в AmoConstruct, есть метод createContacts, создаем метод у сущности контакт createContacts() и либо при создании объекта запоминаем в свойствах контроллер, который он будет использовать, либо передаем контроллер в метод createContacts(AmoControl $obj) {$obj->createContacts(param...)}

Felotus committed:
18.01 ашкые

Сделал 1 точку, нужно только view адекватно построить, сейчас займусь проверкой всего, правками

Felotus committed:
18.01 first

Сделал 1 точку, нужно только view адекватно построить, сейчас займусь проверкой всего, правками

Felotus committed:
18.01 first

Сделал 1 точку, нужно только view адекватно построить, сейчас займусь проверкой всего, правками

Felotus committed:
18.01 last

Внес логические правки, осталось кодстайл, может еще View нормальные сделать.

Felotus committed:
18.01 last

Внес логические правки, осталось кодстайл, может еще View нормальные сделать.

Felotus committed:
18.01 Last

Внес логические правки, осталось кодстайл, может еще View нормальные сделать.
